chingu-voyages/v40-geckos-team-20
# Feature/cocktail list error loading states - EC-59, EC-38, EC-39

@hmshp please review this PR

It covers the following tickets:
https://viteshbava.atlassian.net/browse/EC-59
https://viteshbava.atlassian.net/browse/EC-38
https://viteshbava.atlassian.net/browse/EC-39

This PR introduces a loading state, error state, and no results state for the Cocktail List.

How to test:
* Loading state - easiest way is to just refresh the page and check you can see the spinner while the random cocktails are being pulled from the api.
* Error state - in .src/api/cocktailApi.js, temporarily change either `BASE_URL`, `URL.SEARCH` or `URL.RANDOM` so that it is misspelled, and will therefore cause an error when loading the page or performing a search.  Then check that the error state appears correctly.
* No results state - do a search for something that you know should not return any results, and check the no results state appears correctly.
* If you can think of any other test or something I've missed or overlooked, please test / let me know.

If you have any questions or concerns about the code I have changed please let me know.

Thanks!
